tests/e2e/test_couchbase.py

```diff
@@ -0,0 +1,328 @@
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.ocs.utils import get_pod_name_by_pattern
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import E2ETest, workloads
+from ocs_ci.ocs.exceptions import CommandFailed
+
+log = logging.getLogger(__name__)
+
+
+@workloads
+class TestCouchbaseWorkload(E2ETest):
+    """
+    Main couchbase workload class
+    """
+    MIN_ACCEPTABLE_OPS_PER_SEC = 1000
+    MAX_ACCEPTABLE_RESPONSE_TIME = 1000
+    COUCHBASE_OPERATOR = 'couchbase-operator-namespace'
+    DOCKER_ENV_VAR_USERNAME = 'CBRHELUSER'
+    DOCKER_ENV_VAR_PASSWORD = 'CBRHELPASSWORD'
+    DOCKER_ENV_VAR_EMAIL = "CBRHELEMAIL"
+    admission_parts = [
+        constants.COUCHBASE_ADMISSION_SERVICE_ACCOUNT_YAML,
+        constants.COUCHBASE_ADMISSION_CLUSTER_ROLE_YAML,
+        constants.COUCHBASE_ADMISSION_CLUSTER_ROLE_BINDING_YAML,
+        constants.COUCHBASE_ADMISSION_SECRET_YAML,
+        constants.COUCHBASE_ADMISSION_DEPLOYMENT_YAML,
+        constants.COUCHBASE_ADMISSION_SERVICE_YAML,
+        constants.COUCHBASE_MUTATING_WEBHOOK_YAML,
+        constants.COUCHBASE_VALIDATING_WEBHOOK_YAML
+    ]
+    admission_objs = []
+    pod_obj = OCP(kind='pod')
+    couchbase_pod = OCP(kind='pod')
+    admission_pod = []
+    secrets_array = []
+    secret_options = [
+        ["docker-username", DOCKER_ENV_VAR_USERNAME],
+        ["docker-password", DOCKER_ENV_VAR_PASSWORD],
+        ["docker-email", DOCKER_ENV_VAR_EMAIL]
+    ]
+
+    def gen_secret(self, entry):
+        """
+        Each entry consists of a parameter name and an environment variable.
+        Generate text to add to the docker command if the environment variable
+        is set.
+        """
+        envar = os.environ.get(entry[1])
+        if envar:
+            return f" --{entry[0]}={envar}"
+        return False
+
+    def create_secret(self):
+        """
+        Create a docker-registry secret.  This call uses environment variables
+        CBRHELUSER for the username, CBRHELPASSWORD for the password, and
+        CBRHELEMAIL for the email address.  If any of these values are not
+        set, create_secret still attempts to run the create command without
+        these options specified.
+        """
+        docker_cmd = 'create secret docker-registry rh-catalog '
+        docker_cmd += '--docker-server=registry.connect.redhat.com'
+        secret_array = []
+        for entry in self.secret_options:
+            new_create_cmd_line_part = self.gen_secret(entry)
+            if new_create_cmd_line_part:
+                secret_array.append(entry[0])
+                docker_cmd += new_create_cmd_line_part
+        self.pod_obj.exec_oc_cmd(command=docker_cmd, secrets=secret_array)
+
+    def add_serviceaccount_secret(self, in_text):
+        """
+        Execute secrets add for serviceaccount
+        """
+        self.secretsadder.exec_oc_cmd(
+            f"secrets add serviceaccount/{in_text} secrets/rh-catalog --for=pull"
+        )
```

Comment:
how are you making sure that commend run successfully and you have secret added ?
is not it better to have get secret call to make sure its added then exit function ?

Reply:
This code is changing due to changes in how the secrets are generated.  The new version will not require the environment variables used in the older version.

Reply:
This will be fixed in the next checkin